controllers/btpoperator_controller.go

```diff
@@ -1131,3 +1179,567 @@ func (r *BtpOperatorReconciler) watchConfigPredicates() predicate.Funcs {
+	return caCertificate, caPrivateKey, nil
+}
+
+func (r *BtpOperatorReconciler) generateSignedCertAndAddToApplyList(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured, CA, CAPrivateKey []byte) error {
+	logger := log.FromContext(ctx)
+	logger.Info("generation of signed webhook certificate started")
+
+	webhookCertificate, webhookPrivateKey, err := r.generateSignedCert(ctx, time.Now().Add(WebhookCertificateExpiration), CA, CAPrivateKey)
+	if err != nil {
+		return fmt.Errorf("while generating signed webhook certificate: %w", err)
+	}
+	err = r.appendCertificationDataToUnstructured(resourcesToApply, WebhookSecret, webhookCertificate, webhookPrivateKey, WebhookSecretDataPrefix)
+	if err != nil {
+		return fmt.Errorf("while adding newly generated signed webhook certificate to resoruces to apply: %w", err)
+	}
+	logger.Info("generation of signed webhook certificate success")
+	return nil
+}
+
+func (r *BtpOperatorReconciler) generateSignedCert(ctx context.Context, expiration time.Time, caCertificate, caPrivateKey []byte) ([]byte, []byte, error) {
+	if caCertificate == nil || caPrivateKey == nil {
+		data, err := r.getDataFromSecret(ctx, CaSecret)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		caCertificate, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, CertificatePostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+
+		caPrivateKey, err = r.getValueByKey(r.buildKeyNameWithExtension(CaSecretDataPrefix, RsaKeyPostfix), data)
+		if err != nil {
+			return []byte{}, nil, err
+		}
+	}
+
+	webhookCertificate, webhookPrivateKey, err := certs.GenerateSignedCertificate(expiration, caCertificate, caPrivateKey)
+	if err != nil {
+		return []byte{}, nil, err
+	}
+
+	return webhookCertificate, webhookPrivateKey, err
+}
+
+func (r *BtpOperatorReconciler) appendCertificationDataToUnstructured(us *[]*unstructured.Unstructured, certName string, certificate, privateKey []byte, prefix string) error {
+	err, data := r.mapCertToSecretData(certificate, privateKey, r.buildKeyNameWithExtension(prefix, CertificatePostfix), r.buildKeyNameWithExtension(prefix, RsaKeyPostfix))
+	if err != nil {
+		return err
+	}
+	secret := r.buildSecretWithDataAndLabels(certName, data, map[string]string{managedByLabelKey: operatorName})
+
+	unstructuredObj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(secret)
+	if err != nil {
+		return err
+	}
+	*us = append(*us, &unstructured.Unstructured{Object: unstructuredObj})
+	return nil
+}
+
+func (r *BtpOperatorReconciler) mapCertToSecretData(certificate, privateKey []byte, keyNameForCert, keyNameForPrivateKey string) (error, map[string][]byte) {
+	return nil, map[string][]byte{
+		keyNameForCert:       certificate,
+		keyNameForPrivateKey: privateKey,
+	}
+}
+
+func (r *BtpOperatorReconciler) reconcileWebhooksConfigurations(ctx context.Context, resourcesToApply *[]*unstructured.Unstructured, expectedCa []byte) error {
```

Comment:
think about changing the `*[]*unstructured.Unstructured` param to just `[]*unstructured.Unstructured` when there is no need to change the length of the slice. Imo, it could improve the readability of the code